model/riscv_vext_regs.sail

```diff
@@ -0,0 +1,483 @@
+/* vector registers */
+register vr0 : vregtype
+register vr1 : vregtype
+register vr2 : vregtype
+register vr3 : vregtype
+register vr4 : vregtype
+register vr5 : vregtype
+register vr6 : vregtype
+register vr7 : vregtype
+register vr8 : vregtype
+register vr9 : vregtype
+register vr10 : vregtype
+register vr11 : vregtype
+register vr12 : vregtype
+register vr13 : vregtype
+register vr14 : vregtype
+register vr15 : vregtype
+register vr16 : vregtype
+register vr17 : vregtype
+register vr18 : vregtype
+register vr19 : vregtype
+register vr20 : vregtype
+register vr21 : vregtype
+register vr22 : vregtype
+register vr23 : vregtype
+register vr24 : vregtype
+register vr25 : vregtype
+register vr26 : vregtype
+register vr27 : vregtype
+register vr28 : vregtype
+register vr29 : vregtype
+register vr30 : vregtype
+register vr31 : vregtype
+
+val vreg_name : bits(5) <-> string
+mapping vreg_name = {
+    0b00000 <-> "v0",
+    0b00001 <-> "v1",
+    0b00010 <-> "v2",
+    0b00011 <-> "v3",
+    0b00100 <-> "v4",
+    0b00101 <-> "v5",
+    0b00110 <-> "v6",
+    0b00111 <-> "v7",
+    0b01000 <-> "v8",
+    0b01001 <-> "v9",
+    0b01010 <-> "v10",
+    0b01011 <-> "v11",
+    0b01100 <-> "v12",
+    0b01101 <-> "v13",
+    0b01110 <-> "v14",
+    0b01111 <-> "v15",
+    0b10000 <-> "v16",
+    0b10001 <-> "v17",
+    0b10010 <-> "v18",
+    0b10011 <-> "v19",
+    0b10100 <-> "v20",
+    0b10101 <-> "v21",
+    0b10110 <-> "v22",
+    0b10111 <-> "v23",
+    0b11000 <-> "v24",
+    0b11001 <-> "v25",
+    0b11010 <-> "v26",
+    0b11011 <-> "v27",
+    0b11100 <-> "v28",
+    0b11101 <-> "v29",
+    0b11110 <-> "v30",
+    0b11111 <-> "v31"
+}
+
+val rV : forall 'n, 0 <= 'n < 32. regno('n) -> vregtype effect {rreg, escape}
+function rV r = {
+  let zero_vreg : vregtype = EXTZ(0x0);
```

Comment:
The X and F equivalents are global, should this be consistent?

Reply:
When I move this zero_vreg definition as global, it will cause a strange error at compile time. I'll figure it out later.

Reply:
Will be handled later.